src/renderer/shared/ui-kit/hooks/useIntersectionObserver.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react';
+
+const elementCallbacks = new Map<Element, (isIntersecting: boolean) => void>();
+
+const observer = new IntersectionObserver(
+  (entries) => {
+    for (const entry of entries) {
+      const callback = elementCallbacks.get(entry.target);
+      if (callback) {
+        callback(entry.isIntersecting);
+      }
+    }
+  },
+  {
+    rootMargin: '200px',
+    threshold: 0.1,
+  },
+);
+
+export function useIntersectionObserver(element: Element | null) {
```

Comment:
I think `useIntersectionObserver` should subscribe entities from observer similar to `useResizeObserver`. This logic looks more like `useViewportIntersection`, idk